src/main/java/org/skriptlang/skript/bukkit/recipes/RecipeWrapper.java

```diff
@@ -0,0 +1,367 @@
+package org.skriptlang.skript.bukkit.recipes;
+
+import org.bukkit.Bukkit;
+import org.bukkit.NamespacedKey;
+import org.bukkit.inventory.*;
+import org.bukkit.inventory.recipe.CookingBookCategory;
+import org.bukkit.inventory.recipe.CraftingBookCategory;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+public class RecipeWrapper implements Recipe {
+
+	private final RecipeUtils.RecipeType recipeType;
+	private final NamespacedKey key;
+	private ItemStack result = null;
+	private List<String> errors = new ArrayList<>();
+
+	public RecipeWrapper(NamespacedKey key, RecipeUtils.RecipeType recipeType) {
+		this.key = key;
+		this.recipeType = recipeType;
+	}
+
+	public void setResult(ItemStack result) {
+		this.result = result;
+	}
+
+	public ItemStack getResult() {
+		return result;
+	}
+
+	public NamespacedKey getKey() {
+		return this.key;
+	}
+
+	public RecipeUtils.RecipeType getRecipeType() {
+		return recipeType;
+	}
+
+	public void addError(String error) {
+		errors.add(error);
+	}
+
+	public List<String> getErrors() {
+		return errors;
+	}
+
+	public Recipe create() {
+		return null;
+	}
+
+    public static class CraftingRecipeWrapper extends RecipeWrapper {
```

Comment:
shouldn't these implement their own counterparts

Reply:
Do you mean also extending CraftingRecipe? If so, then I guess, I dont think it would make much of a difference maybe

Reply:
well i should be able to treat a mutable crafting recipe the same as a crafting recipe
hence they should have the same interface, and it would allow it to pass `instanceof CraftingRecipe`